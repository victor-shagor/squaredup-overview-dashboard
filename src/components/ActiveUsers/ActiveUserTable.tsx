import * as React from "react";
import { styled } from "@mui/material/styles";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell, { tableCellClasses } from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import Box from "@mui/material/Box";
import Paper from "@mui/material/Paper";
import { ActiveUserTableProps } from "constants/interfaces";

const StyledTableCell = styled(TableCell)(({ theme }) => ({
  [`&.${tableCellClasses.head}`]: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white,
  },
  [`&.${tableCellClasses.body}`]: {
    fontSize: 14,
  },
}));

const StyledTableRow = styled(TableRow)(({ theme }) => ({
  "&:nth-of-type(odd)": {
    backgroundColor: theme.palette.action.hover,
  },
  // hide last border
  "&:last-child td, &:last-child th": {
    border: 0,
  },
}));

const ActiveUsersTable = ({ users }: ActiveUserTableProps) => {
  const tableHeader = ["S/N", "FirstName", "LastName", "Email", "Created"];
  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 700 }} aria-label="customized table">
        <TableHead>
          <TableRow>
            {tableHeader.map((el) => (
              <StyledTableCell key={el} align="left">
                {el}
              </StyledTableCell>
            ))}
          </TableRow>
        </TableHead>
        <TableBody>
          {users.length ? (
            users.map((row, idx) => (
              <StyledTableRow key={row.email}>
                <StyledTableCell component="th" scope="row">
                  {idx + 1}
                </StyledTableCell>
                <StyledTableCell align="left">{row.firstName}</StyledTableCell>
                <StyledTableCell align="left">{row.lastName}</StyledTableCell>
                <StyledTableCell align="left">{row.email}</StyledTableCell>
                <StyledTableCell align="left">{row.created}</StyledTableCell>
              </StyledTableRow>
            ))
          ) : (
            <Box sx={{ minHeight: "18rem" }}>No data yet</Box>
          )}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default ActiveUsersTable;
