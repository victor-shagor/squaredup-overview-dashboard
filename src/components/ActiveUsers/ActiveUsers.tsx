import { ActiveUserProps } from "constants/interfaces";
import { Card } from "pages/overview/overview.styles";
import ActiveUsersTable from "./ActiveUserTable";

const ActiveUsers = ({
  onClickActiveUserToggle,
  activeUserToggle,
  activeUsers,
}: ActiveUserProps) => {
  return (
    <Card
      height={"25rem"}
      width={"96.5%"}
      className="activechart"
      data-testid="active-user-container"
    >
      <div className="details">
        <p>Active Users and Active Admins</p>
        <div className="columnChartToggleDiv">
          <div className="columnChartToggle">
            <button
              data-testid="user-toggle-button"
              className={
                activeUserToggle === "user"
                  ? "activeToggleButton"
                  : "toggleButton"
              }
              onClick={() => onClickActiveUserToggle("user")}
            >
              Users
            </button>
            <button
              data-testid="admin-toggle-button"
              className={
                activeUserToggle === "admin"
                  ? "activeToggleButton"
                  : "toggleButton"
              }
              onClick={() => onClickActiveUserToggle("admin")}
            >
              Admin
            </button>
          </div>
        </div>
      </div>
      <ActiveUsersTable users={activeUsers} />
    </Card>
  );
};

export default ActiveUsers;
