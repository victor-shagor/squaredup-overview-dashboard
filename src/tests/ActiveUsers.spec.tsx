import renderer from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";
import ActiveUsers from "components/ActiveUsers/ActiveUsers";

const mockActiveUser = [
  {
    firstName: "test1",
    lastName: "vandop",
    email: "ddklsdfkd@red.com",
    created: "3 years ago",
  },
  {
    firstName: "Thiago",
    lastName: "Margarida",
    email: "thiago.margarida@gmail.com",
    created: "3 years ago",
  },
];
it("render active user component with the correct style", async () => {
  const onClickActiveUserToggle = jest.fn();
  render(
    <ActiveUsers
      activeUserToggle={"user"}
      activeUsers={mockActiveUser}
      onClickActiveUserToggle={onClickActiveUserToggle}
    />
  );
  expect(screen.getByTestId("active-user-container")).toHaveStyle(
    `height:25rem`
  );
  expect(screen.getByTestId("active-user-container")).toHaveStyle(
    `width:96.5%`
  );
});

it("should change active toggle button when  clicked", async () => {
  const onClickActiveUserToggle = jest.fn();
  render(
    <ActiveUsers
      activeUserToggle={"user"}
      activeUsers={mockActiveUser}
      onClickActiveUserToggle={onClickActiveUserToggle}
    />
  );
  const userToggleButton = screen.getByTestId("user-toggle-button");
  const adminToggleButton = screen.getByTestId("admin-toggle-button");

  expect(userToggleButton).toHaveStyle(`background-color: #2C2C2C`);
  expect(adminToggleButton).toHaveStyle(`background-color: white`);
  5;
  fireEvent.click(adminToggleButton);

  expect(onClickActiveUserToggle).toHaveBeenCalled();
});

it("should reender table correctly", async () => {
  const onClickActiveUserToggle = jest.fn();
  render(
    <ActiveUsers
      activeUserToggle={"user"}
      activeUsers={mockActiveUser}
      onClickActiveUserToggle={onClickActiveUserToggle}
    />
  );
  const tableRow = screen.getByRole("table").querySelectorAll("td");
  const tableHead = screen.getByRole("table").querySelectorAll("th");

  expect(tableHead[0].innerHTML).toEqual("S/N");
  expect(tableRow[0].innerHTML).toEqual("test1");
  expect(tableRow).toHaveLength(8);
  expect(tableHead).toHaveLength(7);
});

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ActiveUsers
        activeUserToggle={"user"}
        activeUsers={mockActiveUser}
        onClickActiveUserToggle={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
