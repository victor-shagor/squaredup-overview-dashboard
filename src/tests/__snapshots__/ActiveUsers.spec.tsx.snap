// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gsnTZi fNokON activechart"
  data-testid="active-user-container"
  height="25rem"
  width="96.5%"
>
  <div
    className="details"
  >
    <p>
      Active Users and Active Admins
    </p>
    <div
      className="columnChartToggleDiv"
    >
      <div
        className="columnChartToggle"
      >
        <button
          className="activeToggleButton"
          data-testid="user-toggle-button"
          onClick={[Function]}
        >
          Users
        </button>
        <button
          className="toggleButton"
          data-testid="admin-toggle-button"
          onClick={[Function]}
        >
          Admin
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="customized table"
      className="MuiTable-root css-1pclhgl-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
            scope="col"
          >
            S/N
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
            scope="col"
          >
            FirstName
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
            scope="col"
          >
            LastName
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
            scope="col"
          >
            Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1wvnaxz-MuiTableCell-root"
            scope="col"
          >
            Created
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root css-1enotpu-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
            scope="row"
          >
            1
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            test1
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            vandop
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            ddklsdfkd@red.com
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            3 years ago
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-1enotpu-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
            scope="row"
          >
            2
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            Thiago
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            Margarida
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            thiago.margarida@gmail.com
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-132a6sb-MuiTableCell-root"
          >
            3 years ago
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
